src/tutorial/index.rst

```diff
@@ -350,66 +343,60 @@ Additionally, we will add the two constants we defined earlier so that they are
 
 To implement the output message agent function we need to read the agents location (``"x"``, ``"y"``) variables and ID, and then set the message's location and ``"id"`` variable.
 
-To read an agent's variables the :func:`FLAMEGPU->getVariable()<template<typename T, unsigned int N> __device__ T flamegpu::DeviceAPI::getVariable(const char(&)[N]) const>` function is used. As you may expect by now, the variable's type must be passed as a template argument, and it's name is the only argument. To read an agent's ID, :func:`FLAMEGPU->getID()<flamegpu::DeviceAPI::getID>` is called, this special function requires no additional arguments.
+To read an agent's variables the :func:`FLAMEGPU->getVariable()<template<typename T, unsigned int N> __device__ T flamegpu::DeviceAPI::getVariable(const char(&)[N]) const>` function is used in C++. As you may expect by now, the variable's type must be passed as a template argument, and it's name is the only argument. To read an agent's ID, :func:`FLAMEGPU->getID()<flamegpu::DeviceAPI::getID>` is called, this special function requires no additional arguments. The *pure* Python implementation uses the same format of appending types to the function name. The functions are accessible via the ``pyflamegpu`` module. E.g. ``pyflamegpu.getVariableInt()`` for an ``int`` type.
 
-Functionality for the message output is accessed via ``FLAMEGPU->message_out``, this object is specialised depending on the output message type originally specified in the :c:macro:`FLAMEGPU_AGENT_FUNCTION<FLAMEGPU_AGENT_FUNCTION>` macro. The spatial 2D specialisation, :class:`flamegpu::MessageSpatial2D::Out`, has two available functions; :func:`setVariable()<template<typename T, unsigned int N> __device__ void flamegpu::MessageBruteForce::Out::setVariable(const char(&)[N], T) const>` which is common to all message output types, and :func:`setLocation()<flamegpu::MessageSpatial2D::Out::setLocation>` which takes two ``float`` arguments specifying the location of the message in 2D space.
+Functionality for the message output is accessed via ``FLAMEGPU->message_out`` (or named ``message_out`` variable in Agent Python), this object is specialised depending on the output message type originally specified in the :c:macro:`FLAMEGPU_AGENT_FUNCTION<FLAMEGPU_AGENT_FUNCTION>` macro (or via the Python type annotation). The spatial 2D specialisation, :class:`flamegpu::MessageSpatial2D::Out`, has two available functions; :func:`setVariable()<template<typename T, unsigned int N> __device__ void flamegpu::MessageBruteForce::Out::setVariable(const char(&)[N], T) const>` which is common to all message output types, and :func:`setLocation()<flamegpu::MessageSpatial2D::Out::setLocation>` which takes two ``float`` arguments specifying the location of the message in 2D space. The *pure* Python equivalents are of the same format as in other places (e.g. ``setVariableInt`` for the `int` type).
```

Comment:
Code tags require double `` in rst (int)

Reply:
Ha. Thats from switching between rst and github md.